pkgs/top-level/all-packages.nix

```diff
@@ -51,6 +51,11 @@ in
       lib.mapNullable (rs: rs ++ [ bintools ]) (stdenv.allowedRequisites or null);
   };
 
+  stdenvNoLibs = mkStdenvNoLibs stdenv;
+
+  gccStdenvNoLibs = mkStdenvNoLibs gccStdenv;
+  clangStdenvNoLibs = mkStdenvNoLibs clangStdenv;
```

Comment:
This one will still include compiler-rt for -resource-dir, causing infinite recursion when trying to use this for bootstrapping.

Reply:
I would leave that as is, with your #55750 redefining that properly :). (e.g. there should be a `clangStdenvNoLibs` per LLVM, with this picking the default one.)